ckan/views/api.py

```diff
@@ -207,7 +208,8 @@ def mixed(multi_dict: "MultiDict[str, Any]") -> dict[str, Any]:
 
 # View functions
 
-
+# exempt the CKAN API actions from csrf-validation.
+@csrf.exempt
```

Comment:
@wardi @pdelboca It seems like I spoke too soon and got confused by this

If we exempt API calls from CSRF checks we are essentially still allowing cookie based authentication on POST requests to the API

Removing this exemption would prevent POST API calls unless they send the CSRF header (if CSRF protection is on). Would that be enough or do we need to ensure it at a lower level (flask-login maybe)?

Reply:
We need to check the CSRF token on any call (POST or GET) that because e.g. GET can be used to return private user details

Reply:
OK, then we need to drop the first part of this check in JS code (or remove GET from it):

```js
if (!/^(GET|HEAD|OPTIONS|TRACE)$/i.test(settings.type) && !this.crossDomain) {
```

Reply:
maybe this isn't such a big deal. The same information can be retrieved with a GET request to a normal web page without requiring a CSRF token so adding the requirement for our API might be unnecessary.

Reply:
I remember, in one of the dev meetings, we discussed CSRF protection, @wardi  told me that we don't need to include CSRF Protection for the CKAN API actions as they have their own protection from CKAN and that should be enough.

That's why I exempt the CKAN API actions from csrf-validation

Reply:
CSRF attacks rely on cookies being implicitly sent with all requests to a particular domain. Since we do not allow cookie-based authentication for our CKAN API Actions and we are using an access token which we are sending it using Authorization: <token> Header,  we should be secure.
@amercader @wardi

Reply:
but, we *do* allow cookie-based auth for the CKAN API, unless something has changes.

Reply:
Follow-up discussion here: https://github.com/ckan/ckan/issues/7028